src/Backend/Modules/Blog/Engine/Model.php

```diff
@@ -1070,37 +1020,48 @@ public static function update(array $item): int
         return $item['revision_id'];
     }
 
-    /**
-     * Update an existing category
-     *
-     * @param array       $item The new data.
-     * @param array $meta The new meta-data.
-     *
-     * @return int
-     */
-    public static function updateCategory(array $item, array $meta = null): int
-    {
-        // get database
-        $database = BackendModel::getContainer()->get('database');
+    public static function updateCategory(
+        array $item,
+        ?array $meta = null
+    ): void {
+        $entityManager = BackendModel::get(
+            'doctrine.orm.default_entity_manager'
+        );
+        $category = BackendModel::get('blog.repository.category')
+                                ->find($item['id']);
 
-        // update category
-        $updated = $database->update('blog_categories', $item, 'id = ?', [(int) $item['id']]);
+        if (!$category instanceof Category) {
+            return;
+        }
 
-        // meta passed?
-        if ($meta !== null) {
-            // get current category
-            $category = self::getCategory($item['id']);
+        $metaEntity = $category->getMeta();
 
-            // update the meta
-            $database->update('meta', $meta, 'id = ?', [(int) $category['meta_id']]);
+        if ($meta !== null) {
+            $metaEntity->update(
+                $meta['keywords'],
+                (isset($meta['keywords_overwrite'])) ? $meta['keywords_overwrite'] : false,
```

Comment:
same remark as above

Reply:
@carakas I will add this in a future PR where I implement the entity for the Post